### ISSUE TITLE
callbacks e.g. CSV Callback receive constant lr when doing a LR decay

### ISSUE THREAD

**Issue by kretes** [fchollet/keras#4156]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Having a callback in the model:

```
csvc = keras.callbacks.CSVLogger('models/train/results.csv', separator=',', append=True)
```

and model compilation:

```
from keras.optimizers import SGD
sgd = SGD(lr=0.03, decay=0.01)
model.compile(loss='binary_crossentropy',
              optimizer=sgd,
              metrics=['accuracy'])
```

In the final csv in column 'lr' I get always 0.03 instead of lr calculated by decay.

After looking at callbacks.py I think the problem is in the 'logs' object passed to callback not in the callback itself. 

Any hints? I could try to make a PR - given some guidance

**Comment by kgrm** [comment]
The learning rate decay in the Keras' SGD implementation does not change the value of its `lr` variable, it is computed every batch through the iteration number and the global learning rate, which remains unchanged.
